app/uk/gov/hmrc/partnershipidentificationfrontend/api/controllers/JourneyController.scala

```diff
@@ -52,14 +51,7 @@ class JourneyController @Inject()(controllerComponents: ControllerComponents,
   def createLimitedLiabilityPartnershipJourney: Action[JourneyConfig] = createJourney(LimitedLiabilityPartnership)
 
   private def createJourney(partnershipType: PartnershipType): Action[JourneyConfig] =
-    Action.async(parse.json[JourneyConfig] { json =>
-      for {
-        continueUrl <- (json \ continueUrlKey).validate[String]
-        optServiceName <- (json \ optServiceNameKey).validateOpt[String]
-        deskProServiceId <- (json \ deskProServiceIdKey).validate[String]
-        signOutUrl <- (json \ signOutUrlKey).validate[String]
-      } yield JourneyConfig(continueUrl, PageConfig(optServiceName, deskProServiceId, signOutUrl), partnershipType)
-    }) {
+    Action.async(bodyParser = fromJsonToAJourneyConfig(partnershipType)) {
```

Comment:
I'm afraid I think we should revert back to the original way of parsing the request body here. Your method, whether or not an improvement, is a deviation from the current pattern and I'd require the team's agreement before approving the change.